qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1520A.py

```diff
@@ -435,11 +466,12 @@ def __init__(self, parent: 'KeysightB1500', name: Optional[str],
 
     def _cv_sweep_voltages(self) -> Tuple[float, ...]:
         sign = lambda s: s and (1, -1)[s < 0]
-        start_value = self.sweep_start()
-        end_value = self.sweep_end()
-        step_value = self.sweep_steps()
-        if self.sweep_mode() == 2 or self.sweep_mode() == 4:
-            if not sign(start_value) == sign(self.sweep_end()):
+        start_value = self.cv_sweep.sweep_start()
+        end_value = self.cv_sweep.sweep_end()
+        step_value = self.cv_sweep.sweep_steps()
+        mode = self.cv_sweep.sweep_mode()
+        if mode in (2, 4):
+            if not sign(start_value) == sign(end_value):
```

Comment:
is removing duplication of the sweep_* function below this lines a part of this PR?